src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -45,12 +45,12 @@ def dependency(self):
     def path(self):
         return self._dependency.path
 
-    def register_library(self, name: str) -> MaybeGraph:
-        # TODO: use DistInfoResolver to load wheel/egg/pypi dependencies
-        # TODO: https://github.com/databrickslabs/ucx/issues/1642
-        # TODO: https://github.com/databrickslabs/ucx/issues/1643
-        # TODO: https://github.com/databrickslabs/ucx/issues/1640
-        return MaybeGraph(None, [DependencyProblem('not-yet-implemented', f'Library dependency: {name}')])
+    def register_library(self, library: str) -> list[DependencyProblem]:
+        maybe = self._resolver.resolve_library(self.path_lookup, library)
```

Comment:
Don't remove the TODOs, there's still work to be done here, just not as part of this PR